File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-9.java
class Solution {
    int count = 0;
    int result = -1;

    public int kthSmallest(TreeNode root, int k) {
        dfs(root, k);
        return result;
    }

    public boolean dfs(TreeNode node, int k) {
    if (node == null) return false;

    if (dfs(node.left, k)) return true;

    count++;
    if (count == k) {
        result = node.val;
        return true;
    }

    return dfs(node.right, k);
}

}

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-9.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:
        x = False
        if root is None:
            return False
        if root.val == subRoot.val:
            x = self.isSameTree(root , subRoot)
        return x or self.isSubtree(root.left , subRoot) or self.isSubtree(root.right , subRoot)






    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if p is None and q is None:
            return True
        if p is None or q is None:
            return False
        return p.val == q.val and self.isSameTree(p.left , q.left) and self.isSameTree(p.right , q.right)    